React/wbooks/src/app/screens/Login/components/LoginForm/index.js

```diff
@@ -1,69 +1,56 @@
-    }else {
-      this.validateUser();
+class LoginFormContainer extends Component {
+  handleSubmit = () => {
+    if(!this.props.hasErrors){
+      this.props.handleSubmit(this.props.userName, this.props.password);
+      this.props.initialLoading();
     }
   }
 
   render() {
-    if(this.state.isLogged){
-      return <Redirect to='/dashboard'/>
-    }
-    return (
-      <div className={styles.inputContainer}>
-        <div className={styles.dataContainer}>
-          <img src={userIcon} className={styles.icon} alt='userIcon' />
-          <input className={`${styles.input} ${styles.inputText}`} placeholder='Username' onChange={this.handleNameInput}/>
-        </div>
-        <div className={styles.dataContainer}>
-          <img src={passwordIcon} className={styles.icon} alt='passwordIcon' />
-          <input type='password' className={`${styles.input} ${styles.inputText}`} placeholder='Password' onChange={this.handlePasswordInput}/>
-        </div>
-        <div className={styles.loginButton} onClick={this.handleSubmit}>
-          <h1 className={styles.loginText}>Login</h1>
-        </div>
-        <div className={styles.signupContainer}>
-          <Link className={styles.signupText} to='/signup'>
-            Not a member?
-          </Link>
-        </div>
-        {this.state.hasErrors && <AccessError errors={this.state.hasErrors}/>}
-      </div>
-    );
+    return <LoginForm isLogged={this.props.isLogged} hasErrors={this.props.hasErrors} handleUserNameInput={this.props.handleUserNameInput} handlePasswordInput={this.props.handlePasswordInput} handleSubmit={this.handleSubmit}/>
   }
 }
 
-export default LoginForm;
+const validateInput = createSelector(
+  [state => state.session.userName,
+  state => state.session.password,
+  state => state.session.isLogged],
+  (userName, password, isLogged) => {
+    if(userName === '' || password === ''){
+      return 'Ambos campos son requeridos.';
+    }else if(!regexs.isValidEmail(userName)){
+      return 'El email ingresado no es correcto.';
+    }else if(!regexs.isValidPassword(password)){
+      return 'La contraseña ingresada debe tener entre 8 y 52 caracteres, y una letra y numero.';
+    }else if(isLogged === false){
+      return 'Los datos ingresados no figuran en nuestra base de datos.';
+    }else{
+      return '';
+    }
+  })
+
+const mapStateToProps = state => (
+  { isLogged: state.session.user,
+    hasErrors: validateInput(state),
+    userName: state.session.userName,
+    password: state.session.password }
+);
+
+const mapDispatchToProps = (dispatch) => (
+  { handleUserNameInput: event => dispatch(sessionActions.saveUserName(event.target.value)),
+    handlePasswordInput: event => dispatch(sessionActions.savePassword(event.target.value)),
+    handleSubmit: (userName, password) => dispatch(sessionActions.saveSession(userName, password)),
+    initialLoading: () => dispatch(booksActions.fetchBooks()) }
```

Comment:
I'd change this prop name. In fact, it is not an initial loading. It tends to confuse